Fix issue with Curl and space characters in filename

Eliastik opened the pull request:
Fixes #633 

Fixes an issue with recent versions of Curl not handling correctly space characters in URLs
Curl now seems to not accept URLs with spaces : https://github.com/curl/curl/issues/8654

To fix this issue, space characters are replaced by %20

mkllnk commented on git-ftp:
```diff
@@ -478,7 +478,7 @@ upload_file() {
 	local SRC_FILE="$1"
 	local DEST_FILE="${SRC_FILE#$SYNCROOT}"
-	local ENC_DEST_FILE="${DEST_FILE//#/%23}"
+	local ENC_DEST_FILE="$(echo "${DEST_FILE//#/%23}" | sed 's/ /%20/g')"
```
Cool, that should work. But the previous syntax `"${DEST_FILE//#/%23}"` was already replacing one character. It would probably be better if we used the same technique.

mkllnk reviewed (commented):
Thank you for the pull request. Great to have a fix. Would you mind to add a test case to our test suite?

Eliastik commented:
I added a test case for the issue and changed the method for replacing the space characters (the same as used by the previous syntax for ENC_DEST_FILE)

cirosayagueslaso commented:
Thanks for the fix, good job!

rtharston commented:
This is what I've been looking for, thanks!

dfandrich commented:
Nitpick in the description: spaces aren't allowed in URLs, so there's no "correct" way to handle them.

szogyenyid commented:
@mkllnk Is there anything in the way of merging this?

funder7 commented:
Hi @mkllnk, can you proceed with this please? It seems that only your approval is missing 😃

kitsguru commented:
This change does not appear in the current version on master

funder7 commented:
Yep, I had the same problem again on a Linux machine, installing git-ftp from apt... I thought the repository version didn't have the fix, but there is a regression probably.
